rotkehlchen/exchanges/coinbase.py

```diff
@@ -684,6 +686,98 @@ def _process_normal_trade(self, event: dict[str, Any], trades: list[Trade]) -> N
 
         return None
 
+    def _process_advanced_trade(self, event: dict[str, Any], trades: list[Trade]) -> None:
+        """Turns an advanced_trade_fill transaction into a rotki trade and adds it to the list.
+
+        https://docs.cloud.coinbase.com/sign-in-with-coinbase/docs/api-transactions#transaction-types
+        If the coinbase transaction is not a trade related transaction nothing happens.
+        """
+        try:
+            if event['status'] != 'completed':
+                # We only want to deal with completed trades
+                return None
+
+            # Contrary to the Coinbase documentation we will use created_at, and never
+            # payout_at. It seems like payout_at is not actually the time when the
+            # trade is settled. Reports generated by Coinbase use created_at as well
+            if event.get('created_at') is not None:
+                raw_time = event['created_at']
+            else:
+                raw_time = event['payout_at']
+
+            timestamp = deserialize_timestamp_from_date(raw_time, 'iso8601', 'coinbase')
+            trade_type = TradeType.deserialize(event['advanced_trade_fill']['order_side'])
+
+            if trade_type not in (TradeType.BUY, TradeType.SELL):
+                return None
```

Comment:
This part is common to both functions and can be moved to a third function that returns the type, the timestamp and the amount.

I would create a function `_process_coinbase_trade` and based on the type it executes the common logic and passes it to `_process_advanced_trade` or `_process_normal_trade` and it returns the Trade object generated by the respective of them